docs/book/src/reference/formatting-pipeline.md

```diff
@@ -0,0 +1,64 @@
+# The formatting pipeline
+
+## Query matching
+
+The first step is to apply the formatting queries against an input. That
+is, each Tree-sitter query will be applied to the input's syntax tree,
+taking into account the relevant capture names against all matching
+subtrees. The result of this is effectively serialised into a list of
+atomic units -- formatting directives, defined by the matching capture
+names, and leaf node content (see also [`@leaf`](capture-names/general.md#leaf))
+-- along with the necessary metadata to drive the process.
+
+## Post-processing
+
+The list of atoms from the first step are then processed into a
+canonical form. Specifically:
+
+- Processes [scopes](capture-names/scopes.md);
+- Processes [deletions](capture-names/insertion-and-deletion.md#delete);
+- Processes [case modifications](capture-names/modification.md#lower_case--upper_case);
+- Merges runs of whitespace into a single atom -- or removes them
+  entirely, in the presence of an [antispace](capture-names/horizontal-spacing.md#append_antispace--prepend_antispace)
+  -- and sorts some remaining, adjacent atoms (e.g., hardlines always
+  before spaces, etc.).
+
+## Pretty printing
+
+The pretty printer goes through the post-processed atom collection and
```

Comment:
Also wondering if "goes through the processed atom collection" would make more sense.

Reply:
If we do end up not calling this "post-processing," we should also rewrite the sole mention of the "post-processor" in the note on anchors.